deploy/00_resolve_keep_token.ts

```diff
@@ -0,0 +1,28 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { getNamedAccounts, deployments } = hre
```

Comment:
My IDE (which is IntelliJ) shows me the warning:

```Property 'getNamedAccounts' does not exist on type 'HardhatRuntimeEnvironment'.```
(and also same for `deployments` property)

It looks like it does not see the properties of `HardhatRuntimeEnvironment` that is placed in `hardhat-deploy` directory but only those placed in `hardhat`. Because of that the code completion in the IDE doesn't work as expected because it doesn't show proper method names after writing (for example) `hre.` or `deployments.`.

The solution for that is adding and empty `tsconfig.json` file to the main directory, although i'm not sure exactly why it works. As far as i know the `tsconfig.json` itself indicates that the directory is the root of a TypeScript project (not talking about the properties inside the file which of course have other purposes that the file itself).

As I can see in the [hardhat documentation](https://google.com) the `tsconfig.json` is optional so i guess it is not necessary to add it.

Reply:
VSC shows me the same warning. Adding the `tsconfig.json` did the trick. I think we should take a look at [typescript support for hardhat-deploy plugin](https://hardhat.org/plugins/hardhat-deploy.html#typescript-support).

Reply:
9c206b3459accc8b29a668d24393656f5c2bbfe4